src/Illuminate/Foundation/helpers.php

```diff
@@ -484,24 +484,21 @@ function event(...$args)
 
 if (! function_exists('factory')) {
     /**
-     * Create a model factory builder for a given class, name, and amount.
+     * Create a model factory builder for a given class and amount.
      *
-     * @param  dynamic  class|class,name|class,amount|class,name,amount
+     * @param  string  $class
+     * @param  int  $amount
      * @return \Illuminate\Database\Eloquent\FactoryBuilder
      */
-    function factory()
+    function factory($class, $amount = null)
     {
         $factory = app(EloquentFactory::class);
 
-        $arguments = func_get_args();
-
-        if (isset($arguments[1]) && is_string($arguments[1])) {
-            return $factory->of($arguments[0], $arguments[1])->times($arguments[2] ?? null);
-        } elseif (isset($arguments[1])) {
-            return $factory->of($arguments[0])->times($arguments[1]);
+        if (isset($amount) && is_int($amount)) {
```

Comment:
Should be just `if (! is_null(amount)) {`

Reply:
yah, i went back and forth on this.  looks like it's merged already, so feel to submit a PR.